plugins/catalog/src/components/useOwnUser.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UserEntity } from '@backstage/catalog-model';
+import { useAsync } from 'react-use';
+import { identityApiRef, useApi } from '@backstage/core';
+import { catalogApiRef } from '../plugin';
+
+/**
+ * Get the group memberships of the logged-in user.
+ */
+export function useOwnUser(): {
```

Comment:
react-use exports the actual type which is more precise https://github.com/streamich/react-use/blob/master/src/useAsyncFn.ts

You may want to return that one which helps the caller to separate the different states from each other more easily

Reply:
Interesting. Didn't knew that type could be used externally.